### PR TITLE
Fix tick marks on the x axis of time series plots

### PR DESCRIPTION
**Pull request by xylar**
They are now consistent with the calendar being used and they don't disappear for long time series, as they did before.

### CODE REVIEW

**Comment by xylar** [comment]
This should address #136.

**Comment by milenaveneziani** [comment]
@xylar: I just tested this on beta1, and I now get nice xtick labels (see below). Only one question: do you understand why the `axis tight` feature, to tell it ala matlab, is missing now? Not a big deal, though.

iceareacellnh_20170302 beta1 a_wcycl1850s ne30_oec_icg edison_b1850c5_ne30_v0 4
icevolumecellnh_20170302 beta1 a_wcycl1850s ne30_oec_icg edison_b1850c5_ne30_v0 4

**Comment by xylar** [comment]
@milenaveneziani, hold off on merging for a moment and I'll see if I can get the plots "tight" again.

**Comment by xylar** [comment]
So do we want the x axis tight but a little buffer at the top and bottom on the y axis?  Or do we want the y axis tight, too?

**Comment by xylar** [comment]
Here's an example where the x axis is tight but the y axis isn't:
sst_global_gmpas-qu240_b1850c5_ne30_v0 4

**Comment by milenaveneziani** [comment]
I think a buffer on the y axis is fine.

File: mpas_analysis/shared/plot/plotting.py [hunk 1]
```diff
@@ -1,75 +1,125 @@
...
-    ax.xaxis.set_major_locator(mdates.YearLocator())
-    ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y'))
-    ax.xaxis.set_minor_locator(mdates.MonthLocator())
-    fig.autofmt_xdate()
+    ax.xaxis.set_major_locator(FixedLocator(major, maxXTicks))
+    ax.xaxis.set_major_formatter(FuncFormatter(formatterFun))
+
+    plt.setp(ax.get_xticklabels(), rotation=30)
+
+    plt.autoscale(enable=True, axis='x', tight=True)
```
Comment (xylar) [inline mpas_analysis/shared/plot/plotting.py]: @milenaveneziani, here's the command to make the x axis "tight" in case you need it elsewhere.

**Comment by xylar** [comment]
Okay, then make sure the newest version works for the case you plotted above and I think we're ready to merge.

**Comment by milenaveneziani** [comment]
agreed. better with a bit of buffer.

File: mpas_analysis/shared/plot/plotting.py (hunk 1 above)
Comment (milenaveneziani) [inline mpas_analysis/shared/plot/plotting.py]: thanks!

**Pull request merged**